discord/ui/modal.py

```diff
@@ -168,23 +171,48 @@ async def on_error(self, interaction: Interaction[ClientT], error: Exception, /)
         """
         _log.error('Ignoring exception in modal %r:', self, exc_info=error)
 
-    def _refresh(self, interaction: Interaction, components: Sequence[ModalSubmitComponentInteractionDataPayload]) -> None:
+    def _refresh(
+        self,
+        interaction: Interaction,
+        components: Sequence[ModalSubmitComponentInteractionDataPayload],
+        resolved: Optional[ResolvedDataPayload],
+    ) -> None:
         for component in components:
             if component['type'] == 1:
-                self._refresh(interaction, component['components'])
+                self._refresh(interaction, component['components'], resolved)  # type: ignore
             elif component['type'] == 18:
-                self._refresh(interaction, [component['component']])
+                self._refresh(interaction, [component['component']], resolved)  # type: ignore
             else:
-                item = find(lambda i: getattr(i, 'custom_id', None) == component['custom_id'], self.walk_children())  # type: ignore
+                custom_id = component.get('custom_id')
+                if custom_id is None:
+                    continue
+
+                item = find(
+                    lambda i: getattr(i, 'custom_id', None) == custom_id,
+                    self.walk_children(),
+                )
                 if item is None:
-                    _log.debug('Modal interaction referencing unknown item custom_id %s. Discarding', component['custom_id'])
+                    _log.debug(
+                        'Modal interaction referencing unknown item custom_id: %s. Discarding',
+                        custom_id,
+                    )
                     continue
+
+                # resolved items for selects are sent at once in the modal submit payload
+                if component['type'] in (5, 6, 7, 8):
```

Comment:
```suggestion
                if component['type'] in (5, 6, 7, 8) and resolved is not None:
```
So, idk if I just didn't mimic the PR right, but rn I'm get an error in `_refresh` as resolved is `None`. So this check ensures `resolved` is not `None`. 
To reproduce, submit an emtpy ui.Label + TextInput as the component